cpp/core/jni/JniWrapper.cc

```diff
@@ -856,6 +856,23 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleWriterJniWrapper
     std::shared_ptr<CelebornClient> celebornClient =
         std::make_shared<CelebornClient>(vm, partitionPusher, celebornPushPartitionDataMethod);
     partitionWriterCreator = std::make_shared<CelebornPartitionWriterCreator>(std::move(celebornClient));
+  } else if (partitionWriterType == "uniffle") {
+    shuffleWriterOptions.partition_writer_type = PartitionWriterType::kUniffle;
+    jclass unifflePartitionPusherClass =
+        createGlobalClassReferenceOrError(env, "Lorg/apache/spark/shuffle/writer/PartitionPusher;");
+    jmethodID unifflePushPartitionDataMethod =
+        getMethodIdOrError(env, unifflePartitionPusherClass, "pushPartitionData", "(I[B)I");
+    if (pushBufferMaxSize > 0) {
+      shuffleWriterOptions.push_buffer_max_size = pushBufferMaxSize;
+    }
+    JavaVM* vm;
+    if (env->GetJavaVM(&vm) != JNI_OK) {
+      throw gluten::GlutenException("Unable to get JavaVM instance");
+    }
+    // rename CelebornClient RssClient
+    std::shared_ptr<CelebornClient> celebornClient =
+        std::make_shared<CelebornClient>(vm, partitionPusher, unifflePushPartitionDataMethod);
+    partitionWriterCreator = std::make_shared<CelebornPartitionWriterCreator>(std::move(celebornClient));
```

Comment:
Would you explain a bit why Celeborn's code is reused here?

Reply:
CelebornClient wraps in the partitionWriterCreator mainly do the work related to take out partition data, when `evictPartitions` is called. That will be all the same with different rss framework